### PR TITLE
feat(*): Initial commit

### PR DESCRIPTION
**Pull request by kmala**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,107 @@
-# example-grpc-client
-A simple Go grpc client app for Deis, the open source PaaS https://deis.com
+# Go GRPC Client Quick Start Guide
+
+This guide will walk you through deploying a Go [grpc][grpc] client on [Deis Workflow][]. The client also runs a Go http server which is routable outside of the cluster and proxies the user requests to a Go [grpc server][grpcserver]. The server Go server to which the requests needs to be send can be configured using `deis config:set SERVER_NAME=<server app name>`.
```
Comment (bacongobbler) [inline README.md]: Can you add more explicit docs here saying that to use this demo, you need to deploy example-grpc-server first, then point SERVER_NAME to that app? I don't think a link is sufficient IMO

Comment (bacongobbler) [reply README.md]: Just a link to the grpc server README would suffice since that README contains the demo.

Comment (vdice) [inline README.md]: Typo: Assuming `The server Go server...` is intended to be `The Go server...`

Comment (vdice) [reply README.md]: Also `needs to be send` -> `needs to be sent`

**Comment by vdice** [comment]
Besides typos, code LGTM and tested w/ associated example-grpc-server

File: main.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	http.Handle("/", logRequestMiddleware(http.HandlerFunc(poweredByHandler)))
+	port := getenv("PORT", "5000")
+	log.Printf("listening on %v...\n", port)
+	err := http.ListenAndServe(":"+port, nil)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// poweredByHandler writes a response by qurying the grpc server.
```
Comment (vdice) [inline main.go]: s/qurying/querying

File: glide.lock [hunk 3]
```diff
@@ -0,0 +1,33 @@
+hash: b833c4cda7c9d6b46e59553667191affe6ee81640319606aa981e58f305804d7
+updated: 2016-10-31T16:59:58.336477767-06:00
+imports:
+- name: github.com/golang/protobuf
+  version: 98fa357170587e470c5f27d3c3ea0947b71eb455
+  subpackages:
+  - proto
+- name: github.com/kmala/example-grpc-server
```
Comment (mboersma) [inline glide.lock]: This package and the SHA will need to change once example-grpc-server is merged.

**Review by mboersma — APPROVED** [review]
Tested with example-grpc-server. Looks great, with the caveat I noted below about changing the SHA and package name.

File: main.go [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+		panic(err)
+	}
+}
+
+// poweredByHandler writes a response by querying the grpc server.
+func poweredByHandler(w http.ResponseWriter, r *http.Request) {
+	powered := getenv("POWERED_BY", "Deis")
+	server := getenv("SERVER_NAME", "")
+	if server == "" {
+		log.Println("Server not configured")
```
Comment (bacongobbler) [inline main.go]: can you add here "Server not configured. Please set $SERVER_NAME"

File: main.go [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write([]byte{})
+		return
+	}
+	address := fmt.Sprintf("%s.%s.svc.cluster.local:80", server, server)
```
Comment (bacongobbler) [inline main.go]: is it completely necessary to use the full DNS name along with port 80? I thought `name.name` could be used in most cases. At the very least the `:80` is a bit overkill for this case :)

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: main.go (hunk 5 above)
Comment (kmala) [reply main.go]: changed to use just `name.name`, thought using the full dns name will provide more clarity but port is needed for the grpc to use.

**Review by bacongobbler — APPROVED** [review]

**Pull request merged**